### PR TITLE
cleanup roles api

### PR DESCRIPTION
**Pull request by mattcasey**
I removed the need to pull in the users along with roles when calling `useRoles()` to make it easier to add fake roles for member levels. Whoever reviews this, please search in "files changed" for places i removed "roleups" and double-check my math: `roles` + `members` = `roleups`, in theory :)

I also removed the 'PublicMember' return type - An e2e test failed because we return a different schema for public users, but the useMembers hook didn't reflect this in its types. I tried adding PublicMember to it, but got 51 type errors. It was easier and safer to just add some empty fields and have a single type.

### CODE REVIEW

File: components/[pageId]/DocumentPage/components/BountyProperties/components/BountyReviewers.tsx
Comment (mattcasey) [inline components/[pageId]/DocumentPage/components/BountyProperties/components/BountyReviewers.tsx:1]: this file was no longer used

File: pages/api/roles/index.ts [hunk 1]
```diff
@@ -64,9 +51,26 @@ async function listSpaceRoles(req: NextApiRequest, res: NextApiResponse<ListSpac
     }
   });
 
-  return res
-    .status(200)
-    .json(roles.map((role) => ({ ...role, spaceRolesToRole: !userId ? [] : role.spaceRolesToRole })));
+  const response: ListSpaceRolesResponse[] = roles;
+
+  if (includeMemberLevels) {
```
Comment (mattcasey) [inline pages/api/roles/index.ts:56]: this is not actually used yet... will be in a follow-up PR to fix the UI

File: charmClient/apis/rolesApi.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import type { Role } from '@prisma/client';
+
+import * as http from 'adapters/http';
+import type { ListSpaceRolesResponse } from 'pages/api/roles';
+
+export class RolesApi {
```
Comment (motechFR) [inline charmClient/apis/rolesApi.ts:6]: Good time for this cleanup  👍

File: components/[pageId]/DocumentPage/components/BountyProperties/components/BountyPropertiesHeader.tsx [hunk 3]
```diff
@@ -26,28 +27,28 @@ interface Props {
...
-    !bountyPermissions || !pagePermissions || !roleups
+    !bountyPermissions || !pagePermissions
       ? { hasPermissions: [], missingPermissions: [] }
       : compareBountyPagePermissions({
           bountyPermissions,
           pagePermissions,
           bountyOperations: ['work'],
           pageOperations: ['edit_content'],
-          roleups
+          members
```
Comment (motechFR) [inline components/[pageId]/DocumentPage/components/BountyProperties/components/BountyPropertiesHeader.tsx:51]: Checked across the stack, makes sense

File: components/[pageId]/DocumentPage/components/ProposalProperties.tsx [hunk 4]
```diff
@@ -54,7 +54,7 @@ export default function ProposalProperties({
   const { permissions: proposalFlowFlags, refresh: refreshProposalFlowFlags } = useProposalFlowFlags({ proposalId });
 
   const { members } = useMembers();
-  const { roles = [], roleups } = useRoles();
+  const { roles = [] } = useRoles();
```
Comment (motechFR) [inline components/[pageId]/DocumentPage/components/ProposalProperties.tsx:57]: Could we put this inside the hook itself?

File: pages/api/spaces/[id]/members/index.ts [hunk 5]
```diff
@@ -112,75 +115,7 @@ async function getMembers(req: NextApiRequest, res: NextApiResponse<(Member | Pu
...
-
-      if (first < second) {
-        return -1;
-      } else if (second > first) {
-        return 1;
-      } else {
-        return 0;
-      }
-    }); // sort members alphabetically
+  const members = await getSpaceMembers({ spaceId, requestingUserId: userId, whereOr });
```
Comment (motechFR) [inline pages/api/spaces/[id]/members/index.ts:118]: Proper cleanup!

File: lib/members/interfaces.ts [hunk 6]
```diff
@@ -31,18 +31,21 @@ export type PropertyValueDetails = {
...
 
-export type Member = Omit<User, 'addresses'> & {
-  isAdmin: boolean;
+export type Member = Pick<User, 'id' | 'createdAt' | 'updatedAt' | 'username'> & {
+  avatar?: string;
+  isBot?: boolean;
+  path?: string;
+  deletedAt?: Date;
+  avatarTokenId?: string;
+  isAdmin?: boolean;
```
Comment (motechFR) [inline lib/members/interfaces.ts:40]: let's merge my PR for guest roles first since it also introduces isGuest concept

**Review by msadura — APPROVED** [review]

**Pull request merged**